frame/balances/src/tests/mod.rs

```diff
@@ -0,0 +1,299 @@
+	type OperationalFeeMultiplier = ConstU8<5>;
+	type WeightToFee = IdentityFee<u64>;
+	type LengthToFee = IdentityFee<u64>;
+	type FeeMultiplierUpdate = ();
+}
+
+impl Config for Test {
+	type Balance = u64;
+	type DustRemoval = DustTrap;
+	type RuntimeEvent = RuntimeEvent;
+	type ExistentialDeposit = ExistentialDeposit;
+	type AccountStore = TestAccountStore;
+	type MaxLocks = ConstU32<50>;
+	type MaxReserves = ConstU32<2>;
+	type ReserveIdentifier = TestId;
+	type WeightInfo = ();
+	type HoldIdentifier = TestId;
+	type FreezeIdentifier = TestId;
+	type MaxFreezes = ConstU32<2>;
+	type MaxHolds = ConstU32<2>;
+}
+
+#[derive(Clone)]
+pub struct ExtBuilder {
+	existential_deposit: u64,
+	monied: bool,
+	dust_trap: Option<u64>,
+}
+impl Default for ExtBuilder {
+	fn default() -> Self {
+		Self { existential_deposit: 1, monied: false, dust_trap: None }
+	}
+}
+impl ExtBuilder {
+	pub fn existential_deposit(mut self, existential_deposit: u64) -> Self {
+		self.existential_deposit = existential_deposit;
+		self
+	}
+	pub fn monied(mut self, monied: bool) -> Self {
+		self.monied = monied;
+		if self.existential_deposit == 0 {
+			self.existential_deposit = 1;
+		}
+		self
+	}
+	pub fn dust_trap(mut self, account: u64) -> Self {
+		self.dust_trap = Some(account);
+		self
+	}
+	pub fn set_associated_consts(&self) {
+		DUST_TRAP_TARGET.with(|v| v.replace(self.dust_trap));
+		EXISTENTIAL_DEPOSIT.with(|v| v.replace(self.existential_deposit));
+	}
+	pub fn build(self) -> sp_io::TestExternalities {
+		self.set_associated_consts();
+		let mut t = frame_system::GenesisConfig::default().build_storage::<Test>().unwrap();
+		pallet_balances::GenesisConfig::<Test> {
+			balances: if self.monied {
+				vec![
+					(1, 10 * self.existential_deposit),
+					(2, 20 * self.existential_deposit),
+					(3, 30 * self.existential_deposit),
+					(4, 40 * self.existential_deposit),
+					(12, 10 * self.existential_deposit),
+				]
+			} else {
+				vec![]
+			},
+		}
+		.assimilate_storage(&mut t)
+		.unwrap();
+
+		let mut ext = sp_io::TestExternalities::new(t);
+		ext.execute_with(|| System::set_block_number(1));
+		ext
+	}
+	pub fn build_and_execute_with(self, f: impl Fn()) {
+		let other = self.clone();
+		SYSTEM_STORAGE.with(|q| q.replace(false));
+		other.build().execute_with(|| f());
+		SYSTEM_STORAGE.with(|q| q.replace(true));
+		self.build().execute_with(|| f());
+	}
+}
+
+parameter_types! {
+	static DustTrapTarget: Option<u64> = None;
+}
+
+pub struct DustTrap;
+
+impl OnUnbalanced<CreditOf<Test, ()>> for DustTrap {
+	fn on_nonzero_unbalanced(amount: CreditOf<Test, ()>) {
+		match DustTrapTarget::get() {
+			None => drop(amount),
+			Some(a) => {
+				let result = <Balances as fungible::Balanced<_>>::resolve(&a, amount);
+				debug_assert!(result.is_ok());
+			},
+		}
+	}
+}
+
+thread_local! {
+	pub static SYSTEM_STORAGE: sp_std::cell::RefCell<bool>  = sp_std::cell::RefCell::new(false);
+}
+pub fn use_system() -> bool {
+	SYSTEM_STORAGE.with(|q| *q.borrow())
+}
```

Comment:
```suggestion
parameter_types! {
	pub static UseSystem: bool = false;
}

// if UseSystem::get() ...
// UseSystem::set(true);
```

I've tried to communicate this new syntax to you a few times in reviews 😅 not sure if it was not seen, or you prefer to old school syntax?

Reply:
under the hood they it generates the same `thread_local` code.

Reply:
Noted.